cmd/syntactic-code-intel-worker/shared/store.go

```diff
@@ -0,0 +1,162 @@
+package shared
+
+import (
+	"context"
+	"database/sql"
+	"strconv"
+	"time"
+
+	"github.com/keegancsmith/sqlf"
+	"github.com/sourcegraph/log"
+	"github.com/sourcegraph/sourcegraph/internal/authz"
+	"github.com/sourcegraph/sourcegraph/internal/authz/providers"
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/conf/conftypes"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
+	connections "github.com/sourcegraph/sourcegraph/internal/database/connections/live"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbutil"
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+	"github.com/sourcegraph/sourcegraph/internal/workerutil"
+	dbworkerstore "github.com/sourcegraph/sourcegraph/internal/workerutil/dbworker/store"
+)
+
+type recordState string
+
+const (
+	Queued     recordState = "queued"
+	Errored    recordState = "errored"
+	Processing recordState = "processing"
+	Completed  recordState = "completed"
+)
+
+// Unless marked otherwise, the columns in this
+// record have a special meaning assigned to them by
+// the queries dbworker performs. You can read more
+// about the different fields and what they do here:
+// https://sourcegraph.com/docs/dev/background-information/workers#database-backed-stores
+type SyntacticIndexingJob struct {
+	ID int `json:"id"`
+	// Which commit to index (not part of dbworker fields)
+	Commit         string      `json:"commit"`
+	QueuedAt       time.Time   `json:"queuedAt"`
+	State          recordState `json:"state"`
+	FailureMessage *string     `json:"failureMessage"`
+	StartedAt      *time.Time  `json:"startedAt"`
+	FinishedAt     *time.Time  `json:"finishedAt"`
+	ProcessAfter   *time.Time  `json:"processAfter"`
+	NumResets      int         `json:"numResets"`
+	NumFailures    int         `json:"numFailures"`
+	// Which repository id to index (not part of dbworker fields)
+	RepositoryID int `json:"repositoryId"`
+	// Name of repository being indexed (not part of dbworker fields)
+	RepositoryName string `json:"repositoryName"`
+	ShouldReindex  bool   `json:"shouldReindex"`
+	// Which user scheduled this job (not part of dbworker fields)
+	EnqueuerUserID int32 `json:"enqueuerUserID"`
```

Comment:
```suggestion
	State          recordState `json:"state"`
	QueuedAt       time.Time   `json:"queuedAt"`
	StartedAt      *time.Time  `json:"startedAt"`
	FinishedAt     *time.Time  `json:"finishedAt"`
	ProcessAfter   *time.Time  `json:"processAfter"`
	NumResets      int         `json:"numResets"`
	NumFailures    int         `json:"numFailures"`
	FailureMessage *string     `json:"failureMessage"`
	ShouldReindex  bool   `json:"shouldReindex"`
    // The fields below are not part of the standard dbworker fields
 
    RepositoryID int `json:"repositoryId"`
    RepositoryName string `json:"repositoryName"`
	Commit         string      `json:"commit"`
	// Which user scheduled this job
	EnqueuerUserID int32 `json:"enqueuerUserID"`
```

Attempted to rearrange this so that related fields are together and all the non-standard fields are grouped together.

Is the repository name preserved here for aiding with debugging to avoid an extra join?